java/README.md

```diff
@@ -55,3 +55,31 @@
      human-readable.
 
 9. Update the code to insert a document with the regular client. What happens?
+
+## KMIP
+
+**KMIP support in this guide relies on unstable dependencies**.
+
+This guide uses the [`direnv`](https://direnv.net/) tool to manage environment variables. 
+
+Update the `TRUSTSTORE_PASSWORD` environment variable in your `.envrc` file to your global Java TrustStore password.
+
+Run `direnv allow` to add the `TRUSTSTORE_PASSWORD` and `KEYSTORE_PASSWORD` environment variables.
+
+Run `source ./configure_certs.sh` to configure certificates start your unstable KMIP
+test server.
+
+⚠️ | Do not use the provided KMIP server in a production environment.
+:---: | :---
+
+Once the server is running, execute the following command:
+
+    mvn compile exec:java -Dexec.mainClass="com.mongodb.csfle.DataEncryptionKeyCreator"
+
+from the `csfle-examples/java` directory to create a data encryption key.
+
+Once you have created your key, add your key to your `InsertDataWithEncryptedFields` class and run the following command:
```

Comment:
I'm not clear what this means.  Looking at InsertDataWithEncryptedFields.java, I don't see where to add the key. Can you make it more clear?

Reply:
Clarified where to add the key value as suggested